src/shared/components/SecondaryButton/SecondaryButton.tsx

```diff
@@ -1,27 +1,39 @@
-import React from 'react';
-import { useNavigate } from 'react-router-dom';
+import React, { ReactEventHandler } from 'react';
+import { Link } from 'react-router-dom';
 import { Button } from '@mui/material';
+import { RoutesPath } from '../../../components/Router/routesPath';
 import './SecondaryButton.scss';
 
+interface IconTypeProps {
+  className?: string;
+  fontSize?: 'small' | 'medium' | 'large' | 'inherit';
+}
+
 interface SecondaryButtonProps {
-  path: string;
   caption: string;
+  path?: string;
+  clickHandler?: ReactEventHandler;
+  className?: string;
+  buttonSize: 'small' | 'large' | 'medium';
+  icon?: (props: IconTypeProps) => JSX.Element;
 }
 
-export const SecondaryButton = ({ path, caption }: SecondaryButtonProps) => {
-  const navigate = useNavigate();
-  const navigateTo = () => {
-    navigate(path);
-  };
-
+export const SecondaryButton = (props: SecondaryButtonProps) => {
+  const { caption, path, clickHandler, className, buttonSize, icon } = props;
   return (
-    <Button
-      className="secondary-button"
-      size="large"
-      variant="outlined"
-      onClick={navigateTo}
-    >
-      {caption}
+    <Button className="secondary-button" size={buttonSize} variant="outlined">
+      <Link
+        className={`secondary-button-link ${className}`}
+        to={path || RoutesPath.HOME}
```

Comment:
Why back button navigating to home?

Reply:
Changed to current route path

Reply:
just make it as required filed, no need to use current location

Reply:
done